dev/com.ibm.ws.org.apache.cxf.cxf.rt.databinding.jaxb.3.2/src/org/apache/cxf/jaxb/JAXBDataBinding.java

```diff
@@ -392,29 +411,54 @@ public synchronized void initialize(Service service) {
 
             if (col.getXmlSchemas().length > 1) {
                 // someone has already filled in the types
+                if (LOG.isLoggable(Level.FINE)) { // Liberty change begin
+                   LOG.fine("initialize: Got schemas for service: " + (serviceInfo != null ? serviceInfo.getName() : "null"));
+                } // Liberty change end
                 justCheckForJAXBAnnotations(serviceInfo);
                 continue;
             }
 
             boolean schemasFromCache = false;
             Collection<DOMSource> schemas = getSchemas();
             if (schemas == null || schemas.isEmpty()) {
+                if (LOG.isLoggable(Level.FINE)) {  // Liberty change begin
+                   LOG.fine("initialize: Try getting schemas from cache");
+                } // Liberty change end
                 schemas = cachedContextAndSchemas.getSchemas();
                 if (schemas != null) {
+                    if (LOG.isLoggable(Level.FINE)) {  // Liberty change begin
+                       LOG.fine("initialize: Got schemas from Context cache");
+                    } // Liberty change end
                     schemasFromCache = true;
                 }
             } else {
+                if (LOG.isLoggable(Level.FINE)) { // Liberty change begin
```

Comment:
Maybe continue using isLoggableFine instead of checking again?

Reply:
yes, will use boolean